lib/events/filesessions/filestream_test.go

```diff
@@ -169,3 +161,48 @@ func TestCompleteUpload(t *testing.T) {
 		})
 	}
 }
+
+func TestCleanupEmptyUpload(t *testing.T) {
+	ctx := t.Context()
+
+	handler, err := NewHandler(Config{
+		Directory: t.TempDir(),
+		OpenFile:  os.OpenFile,
+	})
+	require.NoError(t, err)
+
+	sessionID := session.NewID()
+
+	// Create a completed upload.
+	upload, err := handler.CreateUpload(ctx, sessionID)
+	require.NoError(t, err)
+
+	err = handler.ReserveUploadPart(ctx, *upload, 1)
+	require.NoError(t, err)
+
+	content := []byte("hello world")
+	part, err := handler.UploadPart(ctx, *upload, 1, bytes.NewReader(content))
+	require.NoError(t, err)
+
+	err = handler.CompleteUpload(ctx, *upload, []events.StreamPart{*part})
+	require.NoError(t, err)
+
+	// Create an empty upload with the same session ID and try to complete it.
+	emptyUpload, err := handler.CreateUpload(ctx, sessionID)
+	require.NoError(t, err)
+
+	err = handler.CompleteUpload(ctx, *emptyUpload, []events.StreamPart{})
+	require.NoError(t, err)
+
+	// The empty upload should be cleaned up without impacting the original completed upload.
+	uploadPath := handler.recordingPath(upload.SessionID)
+	f, err := os.Open(uploadPath)
+	require.NoError(t, err)
+
+	gotContent, err := io.ReadAll(f)
+	require.NoError(t, err)
+	require.Equal(t, content, gotContent)
```

Comment:
Let's take a closer look at this fix since test is failing in CI. 
```        	Error Trace:	/__w/teleport/teleport/lib/events/filesessions/filestream_test.go:204
        	Error:      	Not equal: 
        	            	expected: []byte{0x68, 0x65, 0x6c, 0x6c, 0x6f, 0x20, 0x77, 0x6f, 0x72, 0x6c, 0x64}
        	            	actual  : []byte{}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,3 +1,2 @@
        	            	-([]uint8) (len=11) {
        	            	- 00000000  68 65 6c 6c 6f 20 77 6f  72 6c 64                 |hello world|
        	            	+([]uint8) {
        	            	 }
        	Test:       	TestCleanupEmptyUpload
```